docs/jsdoc.rst

```diff
@@ -0,0 +1,160 @@
+==========
+JSDoc Info
+==========
+
+
+**Frequently used tags**
+
+@desc <some description>
+    This tag provides a general description of the symbol or method.
+
+@func [<functionName>]
+    This tag marks an object as being a function with a given name.
+
+@ignore
+    The ignore tag indicates that a symbol in your code should never
+    appear in the documentation. This tag takes precedence over all
+    other tags.
+
+@memberof module:<parentNamePath\|moduleName>
+
+@memberof <namespace>
+    This tag tells a member symbol that belongs to a parent symbol.
+
+@module [<type>] <moduleName>
+    This tag marks the current file as being its own module. All symbols in the file are assumed to be members of the module unless documented otherwise.
+
+@namespace
+    This tag indicates that an object creates a namespace for its
+    members. You can also write a virtual JSDoc comment that defines a namespace used by your code.
+
+    Note: You can not define more than one namespace in one jsdoc block like this:
+
+    /\*\* 
+
+    \*\@namespace firefly
+
+    \*\@namespace firefly.util
+
+    \*\@namespace firefly.action
+
+    \*/
+
+    The jsdoc will only document firefly.action, the last one in the
+    block. To define more than one namespaces in one file you need to:
+
+    /\*\*\@namespace firefly\*/
+
+    /\*\*\@namespace firefly.action\*/
+
+    /\*\*\@namespace firefly.ui\*/
+
+@param
+
+    This tag requires you to specify the name of the parameter you are
+    documenting. You can also include the parameter's type, enclosed in
+    curly brackets, and a description of the parameter. You can use \|
+    to specify more than one type such as {type1\|type2\|type3}.
+
+@static
+
+    This tag indicates that a symbol is contained within a parent and
+    can be accessed without instantiating the parent.
+
+@summary
+
+    This tag is a shorter version of the full description. It can be
+    added to any doclet.
+
+@typedef [<type>] <namepath>
+
+    This tag is useful for documenting custom types, particularly if you
+    wish to refer to them repeatedly. These types can then be used
+    within other tags expecting a type, such
+    as \@type or \@param.
+
+    NOTE: I used @global in each @typedef object. Thus, the typedef
+    object can be referred throughout the documentation.
+
+**When to use @func**
+
+By default, the jsdoc documents all the exported members by
+export.functionName. To document them as static members of a namespace,
```

Comment:
You can make various code literals typeset in monospace by wrapping the text in double back ticks, like

```rst
export.functionName
```